src/client.ts

```diff
@@ -456,4 +468,21 @@ export class RSPClient {
     removeAllListeners(eventName: string): void {
         this.emitter.removeAllListeners(eventName);
     }
+
+    /**
+     * register client capabilities to the server
+     *
+     * @param capabilities client capabilities to register
+     * @param timeout timeout in milliseconds
+     */
+    registerClientCapabilities(capabilities: any, timeout: number = 2000): Promise<Protocol.ServerCapabilitesResponse> {
```

Comment:
Why wouldn't you type the capabilities here? You can just use ClientCapabilitiesRequest.